src/libsystemd-network/sd-dhcp-server.c

```diff
@@ -598,23 +599,28 @@ static int server_send_forcerenew(sd_dhcp_server *server, be32_t address,
 
 static int parse_request(uint8_t code, uint8_t len, const void *option, void *userdata) {
         DHCPRequest *req = userdata;
+        be32_t tmp;
 
         assert(req);
 
         switch(code) {
         case SD_DHCP_OPTION_IP_ADDRESS_LEASE_TIME:
                 if (len == 4)
-                        req->lifetime = be32toh(*(be32_t*)option);
+                        req->lifetime = unaligned_read_be32(option);
 
                 break;
         case SD_DHCP_OPTION_REQUESTED_IP_ADDRESS:
-                if (len == 4)
-                        req->requested_ip = *(be32_t*)option;
+                if (len == 4) {
+                        memcpy(&tmp, option, sizeof(be32_t));
+                        req->requested_ip = tmp;
```

Comment:
why the indirection via tmp? you can just use memcpy() directly from `option` to `req->requested_ip`